pkg/services/ngalert/api/api_alertmanager.go

```diff
@@ -0,0 +1,361 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/go-openapi/strfmt"
+	apimodels "github.com/grafana/alerting-api/pkg/api"
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/store"
+	"github.com/grafana/grafana/pkg/util"
+	amv2 "github.com/prometheus/alertmanager/api/v2/models"
+)
+
+type AlertmanagerSrv struct {
+	am    Alertmanager
+	store store.AlertingStore
+	log   log.Logger
+}
+
+func (srv AlertmanagerSrv) RouteCreateSilence(c *models.ReqContext, body apimodels.SilenceBody) response.Response {
```

Comment:
As discussed, we're not going to be persisting these silences in the database and instead go directly to the alertmanager. Let's leave these endpoints w/o implementation - I'll mark the others as ditto.

The idea here is that the notifier package would implement the necessary methods for you to call as part of these endpoints.

Reply:
reverted to the mock implementation